### PR TITLE
Cleanup build warnings

### PR DESCRIPTION
**Pull request by seglo**
A mixed bag of changes to address warnings and linting errors in the build.

### CODE REVIEW

File: core/src/main/scala/akka/kafka/RestrictedConsumer.scala [hunk 1]
```diff
@@ -36,6 +37,7 @@ final class RestrictedConsumer(consumer: Consumer[_, _], duration: java.time.Dur
   /**
    * See [[org.apache.kafka.clients.consumer.KafkaConsumer#committed(TopicPartition,java.time.Duration)]]
    */
+  @silent
   def committed(tp: TopicPartition): OffsetAndMetadata = consumer.committed(tp, duration)
```
Comment (seglo) [inline core/src/main/scala/akka/kafka/RestrictedConsumer.scala:41]: Perhaps this should be deprecated as well.

File: testkit/src/main/java/akka/kafka/testkit/internal/KafkaContainerCluster.java [hunk 2]
```diff
@@ -214,11 +204,16 @@ private String readinessCheckScript() {
...
+            + connect
+            + " --describe --topic "
+            + READINESS_CHECK_TOPIC
+            + " | wc -l) > 1 ]] && "
+            + "kafka-topics "
             + connect
             + " --delete --topic "
             + READINESS_CHECK_TOPIC
-            + " || echo \"topic does not exist\" \n";
+            + " \n";
```
Comment (seglo) [inline testkit/src/main/java/akka/kafka/testkit/internal/KafkaContainerCluster.java:216]: Gracefully check that the topic exists before trying to delete it. Previous version polluted stderr when readiness check ran successfully.

**Comment by seglo** [comment]
I didn't look at evicted dependencies, but maybe we could clean that up as well.

File: build.sbt [hunk 3]
```diff
@@ -275,7 +276,7 @@ lazy val clusterSharding = project
   .in(file("./cluster-sharding"))
   .dependsOn(core)
   .enablePlugins(AutomateHeaderPlugin)
-  .disablePlugins(MimaPlugin, SitePlugin) // TODO: re-enable MiMa plugin after first release
+  .disablePlugins(MimaPlugin, SitePlugin)
```
Comment (ennru) [inline build.sbt:279]: I guess we should enable Mima?

File: core/src/main/scala/akka/kafka/RestrictedConsumer.scala (hunk 1 above)
Comment (ennru) [reply core/src/main/scala/akka/kafka/RestrictedConsumer.scala:41]: If `consumer.committed` is deprecated, we should deprecate as well.

Comment (seglo) [reply core/src/main/scala/akka/kafka/RestrictedConsumer.scala:41]: I'll make a follow up PR that adds the replacement and deprecates this instance.

File: build.sbt (hunk 3 above)
Comment (seglo) [reply build.sbt]: :+1:

File: core/src/main/scala/akka/kafka/RestrictedConsumer.scala (hunk 1 above)
Comment (seglo) [reply core/src/main/scala/akka/kafka/RestrictedConsumer.scala:41]: https://github.com/akka/alpakka-kafka/pull/1178

**Pull request merged**